tools/journal_proto_processor.cpp

```diff
@@ -1891,18 +1920,46 @@ void JournalProtoProcessor::ProcessMethodExtension(
   }
   cxx_functions_implementation_[descriptor] += cxx_run_epilog + "}\n\n";
 
-  cs_interface_method_declaration_[descriptor] = Join(
-      {"  [DllImport(dllName           : dll_path,\n"
-       "             EntryPoint        = \"principia__" + name + "\",\n"
-       "             CallingConvention = CallingConvention.Cdecl)]",
-       cs_interface_return_marshal,
-       "internal static extern " + cs_interface_return_type + " " + name + "("},
-      "\n  ");
+  cs_interface_method_declaration_[descriptor] =
+      "  private partial class Symbols {\n";
+  cs_interface_method_declaration_[descriptor] +=
+      Join({"    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]",
+            cs_interface_return_marshal,
+            "public delegate " + cs_interface_return_type + " " + name +
+                "Delegate("},
+           "\n    ");
+  if (!cs_interface_marshalled_parameters.empty()) {
+    cs_interface_method_declaration_[descriptor] +=
+        "\n      " + Join(cs_interface_marshalled_parameters,
+                        /*joiner=*/",\n      ");  // NOLINT
```

Comment:
Bad indent.  Use clang format on the new code.  (But don't mess with the existing code.)